plutus-core/plutus-ir/test/transform/inline/letTypeApp2.golden

```diff
@@ -3,5 +3,8 @@
   (termbind
     (strict) (vardecl idFun (all a (type) (fun a a))) (abs a (type) (lam x a x))
   )
-  (con integer 3)
+  [
+    { idFun (con integer) }
+    [ { idFun (con integer) } [ (lam x (con integer) x) (con integer 3) ] ]
+  ]
```

Comment:
This is fine I think. `k` and `funApp` are inlined in 1 pass.

Reply:
But `k` and `funApp` are all unconditionally inlined? I thought we wanted to test that the latest implementation of the callsite inliner (which processes arguments first before the application) can inline more things in 1 pass. Which tests cover that?

Reply:
After processing the arguments first, a few of the test cases did inline more things, including `letFunConstMulti`, `letOverAppMulti`, and `letTypeAppMultiNotSat`. (See https://github.com/input-output-hk/plutus/pull/5426/commits/70291fc96bae2f4400627ad78891ac3030430ccb)